src/test/java/com/rakovets/course/java/core/practice/lambda_expressions/UniversityTest.java

```diff
@@ -0,0 +1,100 @@
+    static Stream<Arguments> groupingByCourseTestProviderArguments() {
+        return Stream.of(
+                Arguments.of(Map.of(
+                        1, Arrays.asList(students.get(3)),
+                        3, Arrays.asList(students.get(1), students.get(4)),
+                        5, Arrays.asList(students.get(0), students.get(2))))
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("groupingByCourseTestProviderArguments")
+    public void groupingByCourseTest(Map<Integer, List<Student>> expected) {
+        Map<Integer, List<Student>> actual = University.groupingByCourse(students);
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    static Stream<Arguments> getSpecTestProviderArguments() {
+        return Stream.of(
+                Arguments.of(Arrays.asList("Chemistry", "KIPIA"))
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("getSpecTestProviderArguments")
+    public void getSpecTest(List<Student> expected) {
+        List<String> actual = University.getSpec(students);
+        Assertions.assertEquals(expected, actual);
+    }
+
+    @Test
+    static Stream<Arguments> countStudentsInSpecProviderArguments() {
+        return Stream.of(
+                Arguments.of(Map.of("Chemistry", 2l, "KIPIA", 3l)));
+    }
+
+    @ParameterizedTest
+    @MethodSource("countStudentsInSpecProviderArguments")
+    public void countStudentsInSpecTest(Map<String, Long> expected) {
+        Map<String, Long> actual = University.countStudentsInSpec(students);
+        Assertions.assertEquals(expected, actual);
+    }
+
+    @Test
+    static Stream<Arguments> groupingBySpecialtyProviderArguments() {
+        return Stream.of(
+                Arguments.of(Map.of(
+                        "Chemistry", Map.of(3, Arrays.asList(students.get(1), students.get(4))),
+                        "KIPIA", Map.of(1, Arrays.asList(students.get(3)),
+                                5, Arrays.asList(students.get(0), students.get(2)))))
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("groupingBySpecialtyProviderArguments")
+    public void groupingBySpecialtyTest(Map<String, Map<Integer, List<Student>>> expected) {
+        Map<String, Map<Integer, List<Student>>> actual = University.groupingBySpecialty(students);
+        Assertions.assertEquals(expected, actual);
+    }
+
+    @Test
+    static Stream<Arguments> checkStudentsProviderArguments() {
+        return Stream.of(
+                Arguments.of(false, 5, Arrays.asList("KIPIA")),
+                Arguments.of(true, 3, Arrays.asList("KIPIA")),
+                Arguments.of(false, 3, Arrays.asList("Chemistry", "KIPIA"))
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("checkStudentsProviderArguments")
+    public void checkStudentsTest(boolean expected, int course, List<Student> withoutSpec) {
+        boolean actual = University.checkStudents(students, course, withoutSpec);
+        Assertions.assertEquals(expected, actual);
```

Comment:
"Violation: There isn't separating empty lines for GIVEN, WHEN, THEN parts
Solution: Add blank line"